core/src/main/scala/org/genivi/sota/core/CampaignResource.scala

```diff
@@ -12,69 +12,97 @@ import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive1, Rout
 import akka.http.scaladsl.unmarshalling._
 import akka.http.scaladsl.util.FastFuture
 import akka.stream.ActorMaterializer
+import cats.Foldable
+import cats.implicits._
 import io.circe.generic.auto._
-import org.genivi.sota.core.data.Campaign
+import org.genivi.sota.core.data.{Campaign, UpdateRequest}
 import org.genivi.sota.core.db.Campaigns
-import org.genivi.sota.data.{Namespace, PackageId}
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.data.{Device, Namespace, PackageId, Uuid}
 import org.genivi.sota.http.ErrorHandler
+import org.genivi.sota.http.UuidDirectives.extractUuid
 import org.genivi.sota.marshalling.CirceMarshallingSupport._
 import org.genivi.sota.rest.Validation.{refined}
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 import slick.driver.MySQLDriver.api.Database
 
-class CampaignResource(db: Database, namespaceExtractor: Directive1[Namespace])
+class CampaignResource(db: Database, namespaceExtractor: Directive1[Namespace],
+                       deviceRegistry: DeviceRegistry, updateService: UpdateService)
                       (implicit system: ActorSystem, mat: ActorMaterializer) {
 
   import StatusCodes.{Success => _, _}
   import Campaign._
+  import UpdateService._
 
   def createCampaign(ns: Namespace, name: CreateCampaign)
                     (implicit ec: ExecutionContext): Route = {
     complete(Created -> db.run(Campaigns.create(ns, name.name)))
   }
 
-  def deleteCampaign(id: Campaign.Id)
+  def deleteCampaign(id: Uuid)
```

Comment:
I know this makes it more consitent, but I think the consitency should go the other way, ie, the others should use wrapper types...